flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java

```diff
@@ -434,17 +434,6 @@
 	@Deprecated
 	public static final String YARN_REALLOCATE_FAILED_CONTAINERS = "yarn.reallocate-failed";
 
-	/**
-	 * The maximum number of failed YARN containers before entirely stopping
-	 * the YARN session / job on YARN.
-	 *
-	 * <p>By default, we take the number of initially requested containers.
-	 *
-	 * @deprecated in favor of {@code YarnConfigOptions#MAX_FAILED_CONTAINERS}.
-	 */
-	@Deprecated
-	public static final String YARN_MAX_FAILED_CONTAINERS = "yarn.maximum-failed-containers";
-
```

Comment:
You cannot simply remove static fields from a class which is annotated with `@Public`.

Reply:
OK. Will push a revert commit. Why the CI system give green to the change. IIRC japi checks are integrated.

Reply:
Thanks for the explanation. I will push a commit to revert the removal of `ConfigConstants#YARN_MAX_FAILED_CONTAINERS` and be careful changes in `@Public`.

Besides, I'd like to know the reason we turn off japicmp plugin for `ConfigConstants`.

Reply:
Another thought is that per @aljoscha mentioned before remove an option with no power notify the user about a problematic settings. But well, here the concern is b/w promise provided by `@Public`

Reply:
Turning off the japicmp plugin check for `ConfigConstants` was a mistake.

One could introduce a special `ConfigOption` which throws an exception if being used. That would have the same effect as breaking a program build after upgrading.